src/reference/antora/modules/ROOT/pages/grpc.adoc

```diff
@@ -0,0 +1,154 @@
+For example, the `GrpcHeaders.METHOD_TYPE` header contains a `io.grpc.MethodDescriptor.MethodType` enum value on the server side (inbound gateway) for easier downstream routing.
+Another useful header is a `GrpcHeaders.SERVICE_METHOD` which indicates what gRPC service method was called on the server, or what gRPC service method to call from the client stub.
+
+NOTE: The `GrpcHeaders.SERVICE_METHOD` header on the inbound gateway has a value of the gRPC service method name exactly as it is declared in the Protobuf (see `.proto` example above) and how it is stored into the `io.grpc.MethodDescriptor` of the service definition.
+
+[[grpc-inbound-gateway]]
+== Inbound Gateway for gRPC
+
+The `GrpcInboundGateway` is a `MessagingGatewaySupport` implementation to receive gRPC requests, send messages downstream flow and produce gRPC responses.
+For initialization, the instance of this gateway requires only an abstract gRPC service class implementing `BindableService`, usually generated from Protobuf and comes with a `*ImplBase` class name.
+
+WARNING: Only standard gRPC services are supported: a generated `AsyncService` contract is what `GrpcInboundGateway` logic based on.
+The Reactor and Kotlin-based service generation don't make sense in Spring Integration logic since those types are not exposed anyhow from the gateway definition.
+
+The gateway uses the mentioned `AsyncService` interface to create proxy and intercept gRPC service methods.
+
+The following example demonstrates how to configure a `GrpcInboundGateway`:
+
+[source, java]
+----
+@Bean
+GrpcInboundGateway helloWorldService() {
+    return new GrpcInboundGateway(TestHelloWorldGrpc.TestHelloWorldImplBase.class);
+}
+----
+
+The `GrpcInboundGateway` implements a `BindableService` and exposes a `ServerServiceDefinition` based on the mentioned proxy for an `AsyncService` contract of the gRPC service.
+Therefore, exactly instance of this gateway has to be registered into a `ServerBuilder` and no need in any other `*ImplBase` implementations in the application.
+
+IMPORTANT: With https://spring.io/projects/spring-grpc[Spring gRPC] and its auto-discovery for `BindableService` implementations, the `GrpcInboundGateway` has to be declared as a top-level bean.
+Therefore, Java DSL API like `IntegrationFlow.from(new GrpcInboundGateway(TestHelloWorldGrpc.TestHelloWorldImplBase.class))` is not recommended because such a `BindableService` implementation won't make it visible for respective Spring gRPC infrastructure.
+
+The `GrpcInboundGateway` uses a `sendAndReceiveMessageReactive()` API to interact with the downstream flow and adapts a `Mono` reply to the gRPC `StreamObserver`.
+As mentioned before, the request message payload is exactly a gRPC request message, and it expects a reply in a from of gRPC response message.
+The downstream logic could be type-safe and deal with gRPC messages similar way as if `*ImplBase` would be implemented manually.
```

Comment:
Rewrite this line as:
`The downstream logic can be type-safe and deal with gRPC messages in a similar way as if *ImplBase were implemented manually.`

WDYT?